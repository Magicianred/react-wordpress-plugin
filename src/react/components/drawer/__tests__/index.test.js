import React from 'react'
import { shallow } from 'enzyme'
import Drawer from '../'

// Mocks & Configs
import getPages from '../../pages/config'
import DrawerList from '../list'
// import PageWrapper from '../../pages'

describe('Components/shallow', () => {
  const props = {
    content: getPages(),
  }

  it('renders app component', () => {
    // test withStyles ==> https://github.com/mui-org/material-ui/issues/9266#issuecomment-349447137
    expect(shallow(<Drawer {...props} />).dive()).toMatchSnapshot()
  })

  it('renders DrawerList component with the correct props', () => {
    const wrapper = shallow(<Drawer {...props} />).dive()

    expect(wrapper.find(DrawerList).length).toEqual(1)
    expect(wrapper.find(DrawerList).props().list).toEqual(getPages())
    expect(wrapper.find(DrawerList).props().onClick).toEqual(wrapper.instance().onPageChange)
  })

  // TODO: fix test, it does not render props.content[1] after the change
  // it('renders PageWrapper component with the correct props', async () => {
  //   const wrapper = shallow(<Drawer {...props} />).dive()
  //   const element = wrapper.find(PageWrapper)
  //   expect(element.length).toEqual(1)
  //   expect(element.props().component).toEqual(props.content[0])
  //
  //   const instance = wrapper.instance()
  //   instance.onPageChange(1)
  //   await nextTick()
  //   expect(element.props().component).toEqual(props.content[1])
  // })

  it('handles onPageChange properly', () => {
    const wrapper = shallow(<Drawer {...props} />).dive()
    const instance = wrapper.instance()
    expect(instance.state).toEqual({ open: false, pageIndex: 0 })

    instance.onPageChange(1)
    expect(instance.state).toEqual({ open: false, pageIndex: 1 })
  })

  it('handles Drawer interactions properly', () => {
    const wrapper = shallow(<Drawer {...props} />).dive()
    const instance = wrapper.instance()
    expect(instance.state).toMatchObject({ open: false })

    instance.handleDrawerOpen()
    expect(instance.state).toMatchObject({ open: true })

    instance.handleDrawerClose()
    expect(instance.state).toMatchObject({ open: false })
  })
})
