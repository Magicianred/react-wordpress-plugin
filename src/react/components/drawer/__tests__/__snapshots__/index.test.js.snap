// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/shallow renders app component 1`] = `
<div
  className="PersistentDrawer-root-1 app-plugin-name"
>
  <div
    className="PersistentDrawer-appFrame-2"
  >
    <WithStyles(AppBar)
      className="PersistentDrawer-appBar-3"
    >
      <WithStyles(Toolbar)
        disableGutters={true}
      >
        <WithStyles(IconButton)
          aria-label="open drawer"
          className="PersistentDrawer-menuButton-6"
          color="inherit"
          onClick={[Function]}
        >
          <pure(Menu) />
        </WithStyles(IconButton)>
        <WithStyles(Typography)
          color="inherit"
          noWrap={true}
          variant="title"
        >
          <FormattedMessage
            defaultMessage="PLUGIN-NAME Settings"
            id="react.settings.title"
            values={Object {}}
          />
        </WithStyles(Typography)>
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
    <WithStyles(Drawer)
      anchor="left"
      classes={
        Object {
          "paper": "PersistentDrawer-drawerPaper-8",
        }
      }
      open={false}
      variant="persistent"
    >
      <div
        className="PersistentDrawer-drawerHeader-9"
      >
        <WithStyles(IconButton)
          onClick={[Function]}
        >
          <pure(ChevronLeft) />
        </WithStyles(IconButton)>
      </div>
      <WithStyles(Divider) />
      <WithStyles(List)>
        <DrawerList
          list={
            Array [
              Object {
                "component": [Function],
                "icon": [Function],
                "name": "Manage",
                "props": undefined,
              },
              Object {
                "component": [Function],
                "icon": [Function],
                "name": "Settings",
                "props": undefined,
              },
            ]
          }
          onClick={[Function]}
        />
      </WithStyles(List)>
    </WithStyles(Drawer)>
    <WithStyles(PageWrapper)
      className="PersistentDrawer-content-10 PersistentDrawer-content-left-11"
      component={
        Object {
          "component": [Function],
          "icon": [Function],
          "name": "Manage",
          "props": undefined,
        }
      }
      placeholder={[Function]}
    />
  </div>
</div>
`;
